Extension/src/LanguageServer/client.ts

```diff
@@ -745,6 +751,46 @@ class DefaultClient implements Client {
         }
     }
 
+    private promptCompileCommands(params: CompileCommandsPaths) : void {
+        if (this.configuration.Configurations[this.configuration.CurrentConfiguration].compileCommands !== undefined) {
+            return;
+        }
+
+        let showCompileCommandsSelection: PersistentState<boolean> = new PersistentState<boolean>("CPP.showCompileCommandsSelection", true);
+        if (!showCompileCommandsSelection.Value) {
+            return;
+        }
+
+        let compileCommandStr: string = params.paths.length > 1 ? "a compile_commands.json file" : params.paths[0];
+        let folderStr: string = vscode.workspace.workspaceFolders.length > 1 ? "the " + this.Name : "this";
```

Comment:
>vscode.workspace.workspaceFolders [](start = 32, length = 33)

should always check that vscode.workspace.workspaceFolders is defined before asking for .length